MaterialLib/SolidModels/Ehlers-impl.h

```diff
@@ -560,6 +547,107 @@ typename SolidEhlers<DisplacementDim>::KelvinVector predict_sigma(
     return sigma_D - pressure * Invariants::identity2;
 }
 
+template <typename ResidualVector, typename KelvinVector>
+std::tuple<KelvinVector, PlasticStrain<KelvinVector>, double>
+splitSolutionVector(ResidualVector const& solution)
+{
+    static auto const size = KelvinVector::SizeAtCompileTime;
+    return std::forward_as_tuple(
```

Comment:
`{...}` (equal to make_tuple) is around 2% (0.5% std. dev) (total time) (at least measurably) slower.
There is magic for r-values in `forward_as_tuple`.